src/subclass/mod.rs

```diff
@@ -12,28 +12,60 @@
 //! ```rust
 //! #[macro_use]
 //! extern crate glib;
+//! #[macro_use]
+//! extern crate genum_derive;
 //! use glib::prelude::*;
 //! use glib::subclass;
 //! use glib::subclass::prelude::*;
 //!
-//! use std::cell::RefCell;
+//! use std::cell::{Cell, RefCell};
+//!
+//! #[derive(Debug, Eq, PartialEq, Clone, Copy, GEnum)]
+//! #[repr(u32)]
+//! #[genum(type_name = "SimpleObjectAnimal")]
+//! enum Animal {
+//!   Goat = 0,
+//!   #[genum(name = "The Dog")]
+//!   Dog = 1,
+//!   #[genum(name = "The Cat", nick = "chat")]
+//!   Cat = 2,
+//! }
+//!
+//! impl Default for Animal {
+//!     fn default() -> Self {
+//!         Animal::Goat
```

Comment:
This is btw what `#[derive(Default)]` would do already :)

Reply:
Would you use it in such example? Wondering if it isn't more exemplary to keep it explicit.

Reply:
I would derive here. It's an example of GLib thingies, not general Rust :) But I don't mind either way

Reply:
Looks like it's not supported with enum:

```
error[E0665]: `Default` cannot be derived for enums, only structs
  --> src/subclass/mod.rs:21:45
   |
10 | #[derive(Debug, Eq, PartialEq, Clone, Copy, Default, GEnum)]
   |                                             ^^^^^^^
```

Reply:
Curious, which language was it that allowed this and had this behaviour then? Oh well :)